src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,29 +106,106 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
                     LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
 
+                    OpenTelemetryClient.LogSuccessMetrics(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        AuthenticationRequestParameters.AppConfig.ClientId, 
+                        authenticationResult.AuthenticationResultMetadata, 
+                        GetCacheLevel(authenticationResult).ToString());
+
+                    LogMsalSuccessTelemetryToOTel(authenticationResult);
+
                     return authenticationResult;
                 }
                 catch (MsalException ex)
                 {
                     apiEvent.ApiErrorCode = ex.ErrorCode;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+
+                    OpenTelemetryClient.LogToFailureCounter(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        AuthenticationRequestParameters.AppConfig.ClientId, 
+                        ex.ErrorCode);
+                    LogMsalFailedTelemetryToOTel(ex);
                     throw;
                 }
                 catch (Exception ex)
                 {
                     apiEvent.ApiErrorCode = ex.GetType().Name;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+                    OpenTelemetryClient.LogToFailureCounter(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        AuthenticationRequestParameters.AppConfig.ClientId, 
+                        ex.GetType().Name);
+                    LogMsalFailedTelemetryToOTel(ex);
                     throw;
                 }
                 finally
                 {
+                    activity?.Stop();
                     telemetryClients.TrackEvent(telemetryEventDetails);
                 }
             }
         }
 
+        private void LogMsalSuccessTelemetryToOTel(AuthenticationResult authenticationResult)
```

Comment:
This is pretty verbose, and probably not something we'd send to the ID4S store?
is it more for partners?

Reply:
Correct, this is not for ID4S and developers can add an exporter to get this data in their own database. This is all the data we log with ITelemetryClient and was discussed with E&D team.